web/src/lib/components/asset-viewer/activity-viewer.svelte

```diff
@@ -15,6 +15,14 @@
 
   const units: Intl.RelativeTimeFormatUnit[] = ['year', 'month', 'week', 'day', 'hour', 'minute', 'second'];
 
+  const shouldGroup = (currentDate: string, nextDate: string): boolean => {
+    return (
+      luxon.DateTime.fromISO(currentDate).toRelative({ unit: 'hours' }) ===
+        luxon.DateTime.fromISO(nextDate).toRelative({ unit: 'hours' }) ||
+      luxon.DateTime.fromISO(currentDate).toRelative() === luxon.DateTime.fromISO(nextDate).toRelative()
```

Comment:
I'm not 100% sure I understand this correctly. If what you're trying to achieve is to check if currentDate and nextData are equal concerning a specific unit, I think you would want `luxon.DateTime.fromISO(currentDate).hasSame(luxonDateTime.fromISO(nextDate), 'hour')`
Also, I think I would prefer to have two variables representing the respective DateTime objects and then use these in the expression.

Reply:
This groups all the comments by units. The issue with `luxon.DateTime.fromISO(currentDate).toRelative() === luxon.DateTime.fromISO(nextDate).toRelative()` is that it correctly groups days, hours, years... But it does not correctly group reactions posted less than an hour ago

Reply:
Ok, I think you could then keep the `currentDateTime.toRelative() === nextDateTime.toRelative()` and the other comparison would be `currentDateTime.hasSame(nextDateTime, 'hour')`